utils/cmake/toolchains/CLANG/compiler-flags.cmake

```diff
@@ -0,0 +1,64 @@
+# See toolchain.cmake before modifying this.
+
+# This file is a copy from ../ARM_GCC, building with Clang/LLVM however will not immediatly work,
+# some flags are not accepted by both compilers (e.g -Wl,--no-wchar-size-warning, defined in target.json), handle conflicts, 
+# and Clang will require the --target=arm-none-eabi flag. We also need Clang to include extra header files:
```

Comment:
Can the `--target=arm-none-eabi` flag be added here?

Reply:
Other flags added to https://github.com/lancaster-university/codal-microbit-v2/compare/master...Johnn333:codal-microbit-v2-clang:master could also be include in this file instead?

Reply:
> Other flags added to [lancaster-university/codal-microbit-v2@master...Johnn333:codal-microbit-v2-clang:master](https://github.com/lancaster-university/codal-microbit-v2/compare/master...Johnn333:codal-microbit-v2-clang:master) could also be include in this file instead?

The main flags added here is for include paths which annoyingly have a version number in their name. Probably not the best thing to push as depending on arm-none-eabi version they may not have the directory? As for the others flags it was mostly to suppress warnings.

Reply:
There is more `-W` flags added to the target-lock.json file. I assume those might not be required and could be removed to reduce the diff?
What was the reason they were added? To help catch issues with the toolchain porting?

Reply:
From memory it was all just to reduce warnings aswell. Clang would give a warning about unused linker inputs so I got rid of them. It still builds but now I think about it may have something to do with the diff? Might be worth looking into.